test/e2e_env/kubernetes/trafficlog/tcp_logging.go

```diff
@@ -0,0 +1,115 @@
+package trafficlog
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/test/e2e_env/kubernetes/env"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/deployments/externalservice"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+)
+
+func TCPLogging() {
+	Describe("TrafficLog Logging to TCP", func() {
+		meshName := "trafficlog-tcp-logging"
+		namespace := "trafficlog-tcp-logging"
+		tcpSinkNamespace := "externalservice-namespace"
+		testServer := "test-server"
+		loggingBackend := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: %s
+`, meshName) + `spec:
+  logging:
+    defaultBackend: netcat
+    backends:
+    - name: netcat
+      format: '%START_TIME(%s)%,%KUMA_SOURCE_SERVICE%,%KUMA_DESTINATION_SERVICE%'
+      type: tcp
+      conf:
+        address: externalservice-tcp-sink.externalservice-namespace.svc.cluster.local:9999
+`
+
+		trafficLog := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: TrafficLog
+mesh: %s
+metadata:
+  name: all-traffic
+spec:
+  sources:
+  - match:
+      kuma.io/service: "*"
+  destinations:
+   - match:
+      kuma.io/service: "*"
+`, meshName)
+		BeforeAll(func() {
+			err := NewClusterSetup().
+				Install(NamespaceWithSidecarInjection(namespace)).
+				Install(MeshKubernetes(meshName)).
+				Install(testserver.Install(
+					testserver.WithNamespace(namespace),
+					testserver.WithMesh(meshName),
+					testserver.WithName(testServer))).
+				Install(DemoClientK8s(meshName, namespace)).
+				Install(externalservice.Install(externalservice.TcpSink, []string{})).
+				Install(YamlK8s(loggingBackend)).
+				Install(YamlK8s(trafficLog)).
+				Setup(env.Cluster)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		E2EAfterAll(func() {
+			Expect(env.Cluster.DeleteDeployment("externalservice-tcp-sink")).To(Succeed())
```

Comment:
this also deletes `externalservice-namespace` and waits until it's deleted, which we want to avoid in e2e_env, because it takes forever.